src/main/java/com/wojciechbarwinski/demo/epic_board_games_shop/controllers/OrderController.java

```diff
@@ -24,4 +27,40 @@ OrderResponseDTO createOrder(@Valid @RequestBody CreateOrderRequestDTO createOrd
 
         return orderServiceFacade.orderProceed(createOrderRequestDto);
     }
+
+    @GetMapping("/orders")
+    List<OrderResponseDTO> getOrders(@RequestParam(value = "page", defaultValue = "0") int page,
+                                @RequestParam(value = "size", defaultValue = "5") int size,
+                                @RequestParam(value = "sortField", required = false) String sortField,
+                                @RequestParam(value = "sortDirection", required = false) SortDirection sortDirection,
+                                @RequestParam(value = "ordererMail", required = false) List<String> ordererMail,
+                                @RequestParam(value = "status", required = false) List<String> status,
+                                @RequestParam(value = "phone", required = false) List<String> phone,
+                                @RequestParam(value = "city", required = false) List<String> city,
+                                @RequestParam(value = "street", required = false) List<String> street,
+                                @RequestParam(value = "sellerMail", required = false) List<String> sellerMail) {
+
+        throwExceptionIfPageOrSizeHasNegativeValue(page, size);
```

Comment:
@RequestParams też można walidować adnotacją @Valid innymi jakby co ;)
ale tak jest git też jak dla mnie

Reply:
Jak się zdzwonimy pokażę Ci jeszcze inny ciekawy sposób jak można to ograć ;)